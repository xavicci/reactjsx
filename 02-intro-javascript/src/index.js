const persona = {
    nombre: 'Tony',
    apellido: 'Stark',
    edad: 45,
    direccion: {
        ciudad: 'New York',
        zip: 5664965,
        lat: 14.235,
        lng: 546.549,
    }
};

// console.table(persona);


const persona2 = { ...persona }
persona2.nombre = 'Peter';

console.log(persona);
console.log(persona2);