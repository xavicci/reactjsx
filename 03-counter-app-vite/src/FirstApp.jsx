

export const FirstApp = () => {
    return (
        <div>
            Hola soy Xavier Flores C
        </div>
    )
}


