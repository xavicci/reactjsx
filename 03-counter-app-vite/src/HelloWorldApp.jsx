export const HelloWorldApp = () => {

    return (
        <div>Hola Mundo !!!</div>
    );
}


