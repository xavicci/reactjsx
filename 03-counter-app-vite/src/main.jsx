import React from 'react';
import ReactDOM from 'react-dom/client';
// import { HelloWorldApp } from './HelloWorldApp';
// import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';



ReactDOM.createRoot(document.getElementById('root')).render(
    // <React.StrictMode>
    // <FirstApp title='hola soy vegeta' />
    <CounterApp value={100} />
    // {/* </React.StrictMode> */}
);