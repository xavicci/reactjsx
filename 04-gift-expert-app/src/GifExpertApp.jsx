import { useState } from 'react'
import { AddCategory, GifGrid } from './components';


export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);

    const onAddCategory = (newCategory) => {

        if (categories.includes(newCategory)) return;
        // categories.push(newCategory);
        // console.log(newCategory);
        setCategories([newCategory, ...categories]);
    };

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory
                onNewCategory={onAddCategory}
            />

            {categories.map((category) => {

                return <GifGrid
                    key={category}
                    category={category} />
            })
            }
        </>
    )
};