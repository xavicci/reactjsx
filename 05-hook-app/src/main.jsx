import React from 'react';
import ReactDOM from 'react-dom/client';

// import { HooksApp } from './HooksApp.jsx';
// import { CounterApp } from './01-useState/CounterApp.jsx';
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook.jsx';
// import { SimpleForm } from './02-useEffect/SimpleForm.jsx';
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook.jsx';
import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks.jsx';

import './index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
  <MultipleCustomHooks />
  // </React.StrictMode>
)
