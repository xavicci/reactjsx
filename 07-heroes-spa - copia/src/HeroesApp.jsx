import { AppRouter } from "./router/AppRouter"

export const HeroesApp = () => {

    return (
        <div>
            <AppRouter />
        </div>
    )
}
