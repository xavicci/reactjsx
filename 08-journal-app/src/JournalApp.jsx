
export const JournalApp = () => {
    return (
        <>
            <h1>HOLA MUNDO</h1>
        </>
    )
}

