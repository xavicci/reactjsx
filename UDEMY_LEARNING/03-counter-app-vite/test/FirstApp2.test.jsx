import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => {
    const title = 'Hola, Soy Goku';
    const subTitle = 'Soy un subtitulo';

    test('debe de hacer match con el snapshot ', () => {

        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostar el mensaje "hola, Soy Goku"', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
        // screen.debug();
    })

    test('debe de mostrar el titulo en un h1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    })

    test('debe de mostar el subtitulo enviado por props', () => {

        render(<FirstApp title={title} subTitle={subTitle} />);
        expect(screen.getAllByText(subTitle).length).toBe(1);
    })

});