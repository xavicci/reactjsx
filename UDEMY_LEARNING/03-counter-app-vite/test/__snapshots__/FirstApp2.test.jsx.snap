// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot  1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <h2>
    1321
  </h2>
  <h2>
    No hay nombres
  </h2>
</div>
`;
